app/controllers/users/confirmations_controller.rb

```diff
@@ -4,22 +4,28 @@ class ConfirmationsController < Devise::ConfirmationsController
 
     def confirm
       with_unconfirmed_confirmable do
```

Comment:
The advantage of the instance variable is you can reference it in other methods, such as `process_successful_password_creation` without having to pass it in as an argument. Maybe the solution is to not use a block and rename the `with_unconfirmed_confirmable` method to something like `initialize_user_and_password_form`.